blockend/public-kernel.js

```diff
@@ -89,9 +118,27 @@ function TinyBoard (size = 50, ttl = ONE_HOUR, now = Date.now) {
           if (expiresAt < now()) return 'RantExpired'
         } break
 
-        case TYPE_BUMP:
+        case TYPE_BUMP:{
           // TODO: model stun-lock with diminishing returns
-          break
+          const rant = state[btok(CHAIN)] // rant
+
+          // get who is bumping
+          const hasBumped = rant.bumpedBy.some((key) => btok(key) === btok(block.key))
```

Comment:
if you want minor speed perf improvement you can use buffer.equals(otherbuffer): `key.equals(block.key)`.
logic is legit